python/src/aiconfig/editor/client/src/components/EditorContainer.tsx

```diff
@@ -172,15 +184,39 @@ export default function EditorContainer({
 
   const onChangePromptName = useCallback(
     async (promptId: string, newName: string) => {
-      const action: AIConfigReducerAction = {
-        type: "UPDATE_PROMPT_NAME",
-        id: promptId,
-        name: newName,
-      };
+      try {
+        const statePrompt = getPrompt(stateRef.current, promptId);
+        if (!statePrompt) {
+          throw new Error(`Could not find prompt with id ${promptId}`);
+        }
+        const prompt = clientPromptToAIConfigPrompt(statePrompt);
 
-      dispatch(action);
+        await debouncedUpdatePrompt(
+          prompt.name,
+          {
+            ...prompt,
+            name: newName,
+          },
+          // PromptName component maintains local state for the name to show in the UI
+          // We cannot update client config state until the name is successfully set server-side
+          // or else we could end up referencing a prompt name that is not set server-side
```

Comment:
Thanks for the great comment!

Reply:
nit: let's add comment of code pointer that this local state happens within the `PromptName` component: https://github.com/lastmile-ai/aiconfig/blob/cec0103bd1f20c7d36b64dcc0249ce6f3817aa21/python/src/aiconfig/editor/client/src/components/prompt/PromptName.tsx#L16

I also have another question, does this dispatch to promptName change happen every time the user changes the name on the keyboard every ~300ms due to DEBOUNCE_MS? https://github.com/lastmile-ai/aiconfig/blob/cec0103bd1f20c7d36b64dcc0249ce6f3817aa21/python/src/aiconfig/editor/client/src/components/prompt/PromptName.tsx#L20

Reply:
The debounce call will happen on each change but the resulting dispatch will happen only after the successful response is returned after the eventual debounced function call (if there's no more input after 300ms)

Reply:
> let's add comment of code pointer that this local state happens within the PromptName component

Is it ok to just keep as-is, with it already referencing `PromptName component`? I think having a code pointer can be brittle since code can easily change to move lines around (e.g. with styling / UI changes)